### PR TITLE
Add typedoc and snippets workflows 

### PR DESCRIPTION
**Pull request by MananTank**
* Add typedoc workflow for generating the full references - for new docs website
* Add Snippets workflow for generating the snippets.json files - for dashboard

`scripts/generate-snippets/*` files are taken from the old docs repo and modified slightly to replace the git submodules usage

### CODE REVIEW

File: scripts/generate-snippets/index.mjs [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+    data.examples = {
+      ...(tsExample?.examples || {}),
+      ...(reactExample?.examples?.javascript
+        ? { react: reactExample.examples.javascript }
+        : {}),
+      ...(pythonExample?.example ? { python: pythonExample.example } : {}),
+      ...(goExample?.example ? { go: goExample.example } : {}),
+    };
+
+    acc[contractName.replace("", "")] = data;
```
[the remark this answers is not in the document]
Comment (MananTank) [reply scripts/generate-snippets/index.mjs]: resolved

File: snippets/readme.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Snippets
+
+This folder contains the following files:
+
+- snippets.json
+- feature_snippets_react.json
+- feature_snippets_sdk.json
```
Comment (joaquim-verges) [inline snippets/readme.md:7]: the unity ones as well?

**Review by joaquim-verges — APPROVED** [review]
lets give it a run with next release?

File: snippets/readme.md (hunk 2 above)
Comment (MananTank) [reply snippets/readme.md:7]: No, it will only generate these 3 files ( same as what we had in docs repo )

**Pull request merged**